File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientDownloadedEpisode.java
package com.mytlogos.enterprise.background.api.model;

public class ClientDownloadedEpisode {
    private final String content;
    private final String title;
    private final int episodeId;

    public ClientDownloadedEpisode(String content, String title, int episodeId) {
        this.content = content;
        this.title = title;
        this.episodeId = episodeId;
    }

    public String getContent() {
        return content;
    }

    public String getTitle() {
        return title;
    }

    public int getEpisodeId() {
        return episodeId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ClientDownloadedEpisode episode = (ClientDownloadedEpisode) o;

        if (getEpisodeId() != episode.getEpisodeId()) return false;
        if (getContent() != null ? !getContent().equals(episode.getContent()) : episode.getContent() != null)
            return false;
        return getTitle() != null ? getTitle().equals(episode.getTitle()) : episode.getTitle() == null;
    }

    @Override
    public int hashCode() {
        int result = getContent() != null ? getContent().hashCode() : 0;
        result = 31 * result + (getTitle() != null ? getTitle().hashCode() : 0);
        result = 31 * result + getEpisodeId();
        return result;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientDownloadedEpisode.kt
package com.mytlogos.enterprise.background.api.model

/**
 * API Model for DownloadContent.
 */
class ClientDownloadedEpisode(private val content: Array<String>, private val title: String, val episodeId: Int) {
    fun getContent(): Array<String> {
        return content
    }

    fun getTitle(): String {
        return title
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val episode = o as ClientDownloadedEpisode
        if (episodeId != episode.episodeId) return false
        if (!getContent().contentEquals(episode.getContent())) return false
        return getTitle() == episode.getTitle()
    }

    override fun hashCode(): Int {
        var result = getContent().contentHashCode()
        result = 31 * result + getTitle().hashCode()
        result = 31 * result + episodeId
        return result
    }
}